optimum/neuron/modeling_decoder.py

```diff
@@ -205,8 +205,12 @@ def _from_pretrained(
             neuronx_model._load_compiled_artifacts(compiled_path)
 
         # Compile the Neuron model (if present compiled artifacts will be reloaded instead of compiled)
-        os.environ["NEURON_CC_FLAGS"] = "--model-type=transformer-inference"
+        neuron_cc_flags = os.environ.get("NEURON_CC_FLAGS", "")
+        os.environ["NEURON_CC_FLAGS"] = (
+            neuron_cc_flags + " --model-type=transformer-inference --enable-experimental-O1"
+        )
```

Comment:
nit: I would use a regex to:

1. Check that `--model-type` was not already set, otherwise we would be setting two values
2. Check that `--enable-experimental-O1` was not already set as well. This is less problematic because it's a flag compared to `--model-type`.

Reply:
Since I add the flags at the end of the existing flags, I actually override model-type, which is absolutely required (compilation will fail if it is not set).
I restore the previous flags afterwards, so it is safe.